### PR TITLE
DataFormats/TrajectorySeed : Changes needed to compile with macOS clang and libc++

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: DataFormats/TrajectorySeed/src/classes_def.xml [hunk 1]
```diff
@@ -13,7 +13,7 @@
   <class name="edm::RefProd<std::vector<TrajectorySeed> >"/>
   <class name="edm::RefVector<std::vector<TrajectorySeed>,TrajectorySeed,edm::refhelper::FindUsingAdvance<std::vector<TrajectorySeed>,TrajectorySeed> >"/>
   <class name="edm::RefVectorIterator<std::vector<TrajectorySeed>,TrajectorySeed,edm::refhelper::FindUsingAdvance<std::vector<TrajectorySeed>,TrajectorySeed> >"/>
-  <class pattern="std::iterator<*edm::Ref*TrajectorySeed*>"/>
+  <!--<class pattern="std::iterator<*edm::Ref*TrajectorySeed*>"/>-->
```
Comment (slava77) [inline DataFormats/TrajectorySeed/src/classes_def.xml]: it would help to have a comment inline with an explanation why this commented out line is needed

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/17353	7ba5f15
- technical; commenting out unused rule; this should eventually be removed
- jenkins tests pass for 91fad21

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by davidlange6** [comment]
+1

**Pull request merged**